### PR TITLE
FAI-10607 - Add run_mode config to support WebhookGaps mode 

### PR DESCRIPTION
**Pull request by matiaslcoulougian**
## Description

Check for  WebhookGaps mode to select streams before records read.

TODO: 
- [ ] Add test case

## Type of change
- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change

## Related issues

> Fix [#1]() 

## Migration notes

> Describe migration notes if any

## Extra info

> Add any additional information

### CODE REVIEW

File: sources/jira-source/src/jira.ts [hunk 1]
```diff
@@ -31,6 +32,7 @@ export interface JiraConfig extends AirbyteConfig {
   readonly cutoff_days?: number;
   readonly cutoff_lag_days?: number;
   readonly board_ids?: ReadonlyArray<string>;
+  readonly run_mode?: RunMode;
```
Comment (ypc-faros) [inline sources/jira-source/src/jira.ts:35]: This needs to be added to the spec. See [here](https://github.com/faros-ai/airbyte-connectors/blob/3bd8555be7373f19f6b109776483442f93785c51/destinations/airbyte-faros-destination/resources/cloud.json#L31) for an example of an enum

File: sources/jira-source/src/streams/common.ts [hunk 2]
```diff
@@ -21,6 +21,17 @@ export type StreamState = {
   readonly [projectOrBoard: string]: {cutoff: number};
 };
 
+export enum RunMode {
+  Full = 'Full',
+  WebhookGaps = 'WebhookGaps',
```
Comment (ypc-faros) [inline sources/jira-source/src/streams/common.ts:26]: nit: I don't love this name. Sounds too negative. What about `WebhookComplement`? cc/ @chalenge

**Comment by ypc-faros** [comment]
Looking good! Let's add the test before merging...

File: sources/jira-source/src/streams/common.ts (hunk 2 above)
Comment (matiaslcoulougian) [reply sources/jira-source/src/streams/common.ts:26]: Like it better, trying to think some other alternatives...

File: sources/jira-source/resources/spec.json
Comment (ypc-faros) [inline sources/jira-source/resources/spec.json:1]: Let's add `"airbyte_secret": true` to user/pass and token

Comment (ypc-faros) [inline sources/jira-source/resources/spec.json:1]: nit: We don't need the `(optional)` in descriptions like `Username for authentication (optional)`
Airbyte properly renders the required and optional fields in the UI based on what's in `required`

File: sources/jira-source/resources/spec.json [hunk 3]
```diff
@@ -119,6 +119,17 @@
...
+      },
+      "run_mode": {
+        "order": 18,
+        "title": "Run Mode",
+        "description": "Run mode for selecting the usage of the connector",
+        "type": "string",
+        "default": "Full",
+        "enum": [
+          "Full",
+          "WebhookGaps"
```
Comment (ypc-faros) [inline sources/jira-source/resources/spec.json:131]: Also rename here => WebhookSupplement

File: sources/jira-source/test/index.test.ts [hunk 4]
```diff
@@ -182,4 +183,24 @@ describe('index', () => {
...
     );
   });
+
+  test('onBeforeRead with run_mode WebhookSupplement should filter streams', async () => {
+    const source = new sut.JiraSource(logger);
+    const catalog = readTestResourceFile('catalog.json');
+    const config = readTestResourceFile('config.json');
+    config.run_mode = RunMode.WebhookSupplement;
+    const {catalog: newCatalog} = await source.onBeforeRead(config, catalog);
+    expect(newCatalog.streams).toHaveLength(3);
```
Comment (ypc-faros) [inline sources/jira-source/test/index.test.ts:193]: ultra nit: Remove this length check. It's redundant given the `toMatchSnapshot` and it's going to be annoying to change it by hand every time we add a stream

**Review by ypc-faros — CHANGES_REQUESTED** [review]
A couple of final comments. We're almost there 🥳

**Review by ypc-faros — APPROVED** [review]

**Pull request merged**